### PR TITLE
Basic fix for tracing data redaction issues

### PR DESCRIPTION
**Pull request by j-baker**
Given that the name is unsafe here, and the name will be correct in the spans on the server, we might get away with just omitting it here?

### CODE REVIEW

**Comment by markelliot** [comment]
I think the fix we originally had made for this on the receiving side was to use the unsubstituted url, since that's known to be safe and conveys a meaningful amount of info.

**Comment by markelliot** [comment]
not sure if that's available at this point in the request chain

**Comment by uschi2000** [comment]
We'd have to implement this at the Feign level, not the OkHttp level.

Sent from Email+ secured by MobileIron

-------- Original Message --------

**Comment by j-baker** [comment]
Cool yeah, so basically talked to @ellisjoe about this and my understanding was: We can't really make much benefit from tracing code at the moment because the logs might contain unsafe information, and both @ellisjoe and @uschi2000 investigated it and couldn't find a 'nice' place that wouldn't involve replicating logic in Feign and Retrofit etc.

Figured we should just merge this pending finding the right place to do it, because tracing > no tracing.

This information should exist on the server side of the span anyway.

**Comment by markelliot** [comment]
More than once have wanted to know at the OkHttp level what the URL template was. Does OkHttp have a way for us to pass that in as metadata on the request object?

On Oct 16, 2017, at 23:06, James Baker <notifications@github.com<mailto:notifications@github.com>> wrote:


Cool yeah, so basically talked to @ellisjoe[github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ellisjoe&d=DwMFaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=Km3h6mvmBmFVuTrOZqmJrOXJ7ge84xKMmE03qNnjNU4&s=CJU8gdrSZRR5DJ-qnzsOhN9pO2xuAtSmmwTN6fAqLzM&e=> about this and my understanding was: We can't really make much benefit from tracing code at the moment because the logs might contain unsafe information, and both @ellisjoe[github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ellisjoe&d=DwMFaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=Km3h6mvmBmFVuTrOZqmJrOXJ7ge84xKMmE03qNnjNU4&s=CJU8gdrSZRR5DJ-qnzsOhN9pO2xuAtSmmwTN6fAqLzM&e=> and @uschi2000[github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_uschi2000&d=DwMFaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=Km3h6mvmBmFVuTrOZqmJrOXJ7ge84xKMmE03qNnjNU4&s=bASVgYmovqg7_SPZP8s-G4m_B777nPsKBFHp_-FIRvM&e=> investigated it and couldn't find a 'nice' place.

Figured we should just merge this pending finding the right place to do it, because tracing > no tracing.

**Comment by uschi2000** [comment]
It does, but as far as I remember you need to unwrap/rewrap a lot of Feign code to access the bit where the OkHttp call is constructed.

**Comment by uschi2000** [comment]
I can probably finish this today or tomorrow.

**Review by uschi2000 — APPROVED** [review]

**Comment by uschi2000** [comment]
Would to really hack Feign (either wrap all proxies or inject&remove HTTP headers) to do the right thing. We're removing the name from the client spans with this PR and rely on the fact that the server span contains the target API name via TraceEnrichingFilter.

**Comment by uschi2000** [comment]
Thanks, James.

File: tracing-okhttp3/src/main/java/com/palantir/remoting3/tracing/okhttp3/OkhttpTraceInterceptor.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
     @Override
     public Response intercept(Chain chain) throws IOException {
         Request request = chain.request();
-        OpenSpan span = Tracer.startSpan(request.method() + " " + request.url(), SpanType.CLIENT_OUTGOING);
+        OpenSpan span = Tracer.startSpan("remote call to redacted url", SpanType.CLIENT_OUTGOING);
```
Comment (gracew) [inline tracing-okhttp3/src/main/java/com/palantir/remoting3/tracing/okhttp3/OkhttpTraceInterceptor.java:35]: I thought in our internal spec we agreed for this to be the method type, e.g. GET, POST, etc.

Comment (uschi2000) [reply tracing-okhttp3/src/main/java/com/palantir/remoting3/tracing/okhttp3/OkhttpTraceInterceptor.java:35]: fixed

**Pull request merged**